src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -551,17 +551,13 @@ private static Details getDetails(DetailAST ast) {
             switch (ast.getType()) {
                 case TokenTypes.LITERAL_TRY:
                 case TokenTypes.LITERAL_CATCH:
-                case TokenTypes.LITERAL_FINALLY:
-                    details = getDetailsForTryCatchFinally(ast);
+                    details = getDetailsForTryCatch(ast);
                     break;
                 case TokenTypes.LITERAL_IF:
-                case TokenTypes.LITERAL_ELSE:
```

Comment:
````
        private static Details getDetailsForIfElse(DetailAST ast) {
            final boolean shouldCheckLastRcurly;
            final DetailAST lcurly;
            DetailAST nextToken = ast.findFirstToken(TokenTypes.LITERAL_ELSE); // always `null`, `else` can't have `else` as child

            if (true) {
                shouldCheckLastRcurly = true;
                nextToken = getNextToken(ast);
                lcurly = ast.getLastChild();
            }
            else {
....
            }

            DetailAST rcurly = null;
            if (lcurly.getType() == TokenTypes.SLIST) {
                rcurly = lcurly.getLastChild();
            }
            return new Details(lcurly, rcurly, nextToken, shouldCheckLastRcurly);
        }
````

````
        private static Details getDetailsForOthers(DetailAST ast) {
            DetailAST rcurly = null;
            final DetailAST lcurly;
            final int tokenType = ast.getType();
            if (false) {
...
            }
            else {
                lcurly = ast.findFirstToken(TokenTypes.SLIST);
                if (lcurly != null) {
                    // SLIST could be absent if method is abstract
                    rcurly = lcurly.getLastChild();
                }
            }
            return new Details(lcurly, rcurly, getNextToken(ast), true);
        }
````

In short, 1 is:
````
shouldCheckLastRcurly = true;
nextToken = getNextToken(ast);
lcurly = ast.getLastChild();
rcurly is `null` or `lcurly.getLastChild()` if lcurly is an slist
````
2 is:
````
lcurly = ast.findFirstToken(TokenTypes.SLIST);
rcurly is `null` or `lcurly.getLastChild()` if lcurly found an slist (same as is an slist)
nextToken = getNextToken(ast);
````

The only difference is lcurly (ast.getLastChild() versus ast.findFirstToken(TokenTypes.SLIST) which is the difference between both being the same if the `else` has a `{` or one will have the next token and the other will have `null`. So it comes down to how we handle `lcurly` in the logic.
Example:
original   - else[75x8] - Details [rcurly=null, lcurly=;[75x12], nextToken=}[76x4], shouldCheckLastRcurly=true]
mutated  - else[75x8] - Details [rcurly=null, lcurly=null, nextToken=}[76x4], shouldCheckLastRcurly=true]

However in this case, it doesn't come down to `lcurly` but `rcurly`.
https://github.com/checkstyle/checkstyle/blob/54616f662130aa0ca5afe45f72d36cdefb78a781/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L308
We don't validate if there is no `rcurly`.

`rcurly` is always `null` if `lcurly` doesn't find an `SLIST`. `rculy` always has the `}` if `lcurly` does find the `SLIST`.

So in the end, they are both the same.